stacks-signer/Cargo.toml

```diff
@@ -25,6 +25,7 @@ clarity = { path = "../clarity" }
 clap = { version = "4.1.1", features = ["derive", "env"] }
 hashbrown = { workspace = true }
 lazy_static = "1.4.0"
+once_cell = "1.8.0"
```

Comment:
Should we bump our rust version and use https://doc.rust-lang.org/std/sync/struct.LazyLock.html instead?

We could also table that for a separate issue. If we do that, I'd prefer that this PR just use `lazy_static!` (so we don't add a new dependency), and then the separate issue would be "replace `lazy_static!` with `std::sync::LazyLock`"

Reply:
seems doable - should this be moved to draft until that separate issue is addressed?

Reply:
There's two options:

1. In this PR, switch to using `std::sync::LazyLock` instead of `once_cell::sync::Lazy` (I think that's pretty much just changing the import). This will make the build only work in rust `1.80`, which might surprise some folks (it is the latest stable release), but I think that's fine.

2. In this PR, switch to using `lazy_static`, because its already a dependency. This would require a bit more code changes, but still pretty minimal. Any question about using 1.80's `LazyLock` support can be punted.

Reply:
I've updated it with the 2nd option so it doesn't stop working for rust versions before 1.80.

Reply:
Another option is to use `std::sync::OnceLock` like this:

```rust
use std::sync::OnceLock;

fn version_string() -> &'static str {
    static VERSION: OnceLock<String> = OnceLock::new();
    VERSION.get_or_init(|| {
        let pkg_version = option_env!("STACKS_NODE_VERSION").unwrap_or(env!("CARGO_PKG_VERSION"));
        let git_branch = GIT_BRANCH.unwrap_or("");
        let git_commit = GIT_COMMIT.unwrap_or("");
        format!(
            "{} ({}:{}, {} build, {} [{}])",
            pkg_version,
            git_branch,
            git_commit,
            BUILD_TYPE,
            std::env::consts::OS,
            std::env::consts::ARCH
        )
    }).as_str()
}
```

No additional crates and will work in Rust 1.70+

Reply:
I've modified to `lazy_static` as that was already imported on stacks-signer before this PR.